### PR TITLE
Set OGRE minimal version to 1.8. Warn on versions not supported (ign-rendering3)

### LINKED ISSUE #375: Provide information about the Ogre version used at configure time
## Desired behavior

Not all the Ogre versions are officially supported in Ignition but most of them can be used to compile the software. It would be useful to provide users with information when they are not using the supported versions.

## Alternatives considered

At least a couple of approaches can be done:

1. Enable unsupported versions using a cmake flag. Fail the build if the versions is not one of the supported ones and implement a flag to enable the configuration to pass with others.

2. Warn users if the ogre version used is not the officially supported one but leave the configuration to finish.

## Implementation suggestion

Not to break people builds out there I think that the safer approach would be to go with 2.

## Additional context

See problems in https://github.com/ignitionrobotics/ign-rendering/issues/374

### PR DESCRIPTION
**Pull request by j-rivero**
Fix #375 

Might require https://github.com/ignitionrobotics/ign-cmake/pull/175 to work

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -61,7 +61,21 @@ endif()
...
+# Ogre versions greater than 1.9 are not officialy supported.
+# Display a warning for the users on this setup unless they provide
+# USE_UNOFFICAL_OGRE_VERSIONS flag
+if (NOT USE_UNOFFICAL_OGRE_VERSIONS)
+  ign_find_package(IgnOGRE VERSION 1.10
+    COMPONENTS ${ign_ogre_components})
+
+  if (OGRE_FOUND)
+    IGN_BUILD_WARNING("Ogre 1.x versions greater than 1.9 are not officially supported."
+                      "The software might compile and event work but support from upstream"
```
Comment (iche033) [inline CMakeLists.txt:73]: `event` - > `even`

> support from upstream could be reduced to accept patches for newer versions

To confirm, I think this means we will provided limited support in the form of accepting patches for newer versions but not actively develop against them right?

File: Changelog.md [hunk 2]
```diff
@@ -1,6 +1,10 @@
 ## Ignition Rendering
 
-### Ignition Rendering 3.X.X (2021-XX-XX)
+### Ignition Rendering 3.x.x (2021-XX-XX)
+
+1. CMake warning on Ogre versions that are not officially supported.
+   To disable the new warning is enough to enable the cmake option USE_UNOFFICAL_OGRE_VERSIONS
```
Comment (iche033) [inline Changelog.md:6]: `To disable the new warning, it is enough to ...`?

File: CMakeLists.txt (hunk 1 above)
Comment (j-rivero) [reply CMakeLists.txt:73]: > To confirm, I think this means we will provided limited support in the form of accepting patches for newer versions but not actively develop against them right?

Eh yes, that is my idea but you Ian or @chapulina probably have it clearer in your mind. Please feel free to reformulate to somehow explain that we won't fix bug on other versions of Ogre but we accept all kind of patches to make it happen.

File: Changelog.md (hunk 2 above)
Comment (j-rivero) [reply Changelog.md]: done 2cdccf2

File: CMakeLists.txt [hunk 3]
```diff
@@ -61,7 +61,21 @@ endif()
...
+                      "could be reduced to accept patches for newer versions")
+  endif()
+endif()
+
+ign_find_package(IgnOGRE VERSION 1.9.0
```
Comment (iche033) [inline CMakeLists.txt:78]: I think we can skip looking for 1.9.0 if 1.10 is found?

File: CMakeLists.txt (hunk 1 above)
Comment (iche033) [reply CMakeLists.txt:73]: Got it. A couple of minor suggestions are: `event` -> `even`, and `accept` -> `accepting`

Comment (j-rivero) [reply CMakeLists.txt]: Done  c57a8e5

File: CMakeLists.txt (hunk 3 above)
Comment (j-rivero) [reply CMakeLists.txt]: how about this 90ebd1e ?

**Review by iche033 — APPROVED** [review]

**Comment by scpeters** [comment]
is this new `USE_UNOFFICAL_OGRE_VERSIONS` in use anywhere yet? can we fix the spelling to `USE_UNOFFICIAL_OGRE_VERSIONS`?

**Comment by j-rivero** [comment]
> is this new `USE_UNOFFICAL_OGRE_VERSIONS` in use anywhere yet? can we fix the spelling to `USE_UNOFFICIAL_OGRE_VERSIONS`?

I did not use it anywhere. It should be fixed, I'll open a PR.

**Comment by traversaro** [comment]
Sorry for being a bit late to the party. : ) However, I was trying to debug some probably related regression in https://github.com/conda-forge/libignition-rendering4-feedstock/pull/19#issuecomment-933532319 , and I was trying to understand how to use the new `USE_UNOFFICIAL_OGRE_VERSIONS` option, and I may be wrong, but it seems that if `USE_UNOFFICIAL_OGRE_VERSIONS` is ON, then `find_package(IgnOgre <..>)` is never invoked? Is that the case, or I am missing something obvious?

**Comment by j-rivero** [comment]
> Sorry for being a bit late to the party. : ) However, I was trying to debug some probably related regression in [conda-forge/libignition-rendering4-feedstock#19 (comment)](https://github.com/conda-forge/libignition-rendering4-feedstock/pull/19#issuecomment-933532319) , and I was trying to understand how to use the new `USE_UNOFFICIAL_OGRE_VERSIONS` option, and I may be wrong, but it seems that if `USE_UNOFFICIAL_OGRE_VERSIONS` is ON, then `find_package(IgnOgre <..>)` is never invoked? Is that the case, or I am missing something obvious?

ouch https://github.com/ignitionrobotics/ign-rendering/pull/453

**Pull request merged**